test/parallel/test-promises-unhandled-rejections.js

```diff
@@ -275,6 +275,20 @@ asyncTest('Attaching a promise catch in a process.nextTick is soon enough to' +
   });
 });
 
+asyncTest('Some weird edge-case', function(done) {
```

Comment:
While inside setImmediate, catching a rejected promise derived from returning a rejected promise in a fulfillment handler prevents unhandledRejection